src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt

```diff
@@ -1,34 +1,116 @@
+    @NotBlank
+    @PathVariable(name = "crn")
+    @Parameter(description = "The identifier of the probation source system (nDelius)", required = true)
+    crn: String,
+  ): List<PersonIdentifierRecord> {
     return handlePersonRecord(personRepository.findByCrn(crn), crn)
   }
 
+  @Operation(description = "Search for person record and associated records with a prisoner number within the system")
   @GetMapping("/search/prisoner/{prisonNumber}")
-  fun searchByPrisonNumber(@NotBlank @PathVariable(name = "prisonNumber") prisonNumber: String): List<PersonIdentifierRecord> {
+  @ApiResponses(
+    ApiResponse(responseCode = "200", description = "OK"),
+    ApiResponse(
+      responseCode = "400",
+      description = "Invalid request.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+    ApiResponse(
+      responseCode = "404",
+      description = "Requested resource not found.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+    ApiResponse(
+      responseCode = "500",
+      description = "Unrecoverable error occurred whilst processing request.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+  )
+  fun searchByPrisonNumber(
+    @NotBlank
+    @PathVariable(name = "prisonNumber")
+    @Parameter(description = "The identifier of the offender source system (NOMIS)", required = true)
+    prisonNumber: String,
+  ): List<PersonIdentifierRecord> {
     return handlePersonRecord(personRepository.findByPrisonNumberAndSourceSystem(prisonNumber), prisonNumber)
   }
 
+  @Operation(description = "Search for person record and associated records with a defendant identifier within the system")
   @GetMapping("/search/defendant/{defendantId}")
-  fun searchByDefendantId(@NotBlank @PathVariable(name = "defendantId") defendantId: String): List<PersonIdentifierRecord> {
+  @ApiResponses(
+    ApiResponse(responseCode = "200", description = "OK"),
+    ApiResponse(
+      responseCode = "400",
+      description = "Invalid request.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+    ApiResponse(
+      responseCode = "404",
+      description = "Requested resource not found.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+    ApiResponse(
+      responseCode = "500",
+      description = "Unrecoverable error occurred whilst processing request.",
+      content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+    ),
+  )
+  fun searchByDefendantId(
+    @NotBlank
+    @PathVariable(name = "defendantId")
+    @Parameter(description = "The identifier of the probation source system (COMMON_PLATFORM)", required = true)
```

Comment:
This is an HMCTS (courts and tribunals) source system rather than a probation one

Reply:
Updated 👍